### PR TITLE
Display answers on summary screen

### PR DESCRIPTION
**Pull request by benjefferies**
### What is the context of this PR?

Describe what you have changed and why, link to other PRs or Issues as appropriate.
[Trello card](https://trello.com/c/bxYkANFg/591-2-rsi-employee-questions-heading-wrong-level)
Changing the quarterly retail survey index schema employee section. The section is now configured as multiple answers to the question `On {date} what was the number of employees?`.
The summary page has changed to support showing multiple answers for a question.
The summary page model had to change to support answers on the summary page
### How to review

Describe the steps required to test the changes (include screenshots if appropriate).
- [ ] Run multiple-answers functional test `npm run test_functional -- --spec=multiple-answers`
- Convert employee questions to answers on survey on rsi 112 survey
- Modify summary.html to display answers on summary
- Add functional test

### CODE REVIEW

File: app/templates/summary.html [hunk 1]
```diff
@@ -22,13 +22,27 @@ <h2 class="header__subtitle mars">Please check carefully before submission.</h2>
 
       {% for item in block.items %}
 
+      {% for answer in item.state_answers %}
+
         {% set id = summary_id ~ "-" ~ loop.index0 %}
 
-        <dt class="summary__question summary__question--{{item.type}}" id="{{id}}">{{item.question | safe}}</dt>
+        <dt class="summary__question summary__question--{{item.type}}" id="{{id}}">
+          {% if item.state_answers|length == 1 %}
```
Comment (iwootten) [inline app/templates/summary.html]: I feel that this kind of conditional logic should be kept away from the frontend templates and instead wrapped up in SummaryItem behaviour.

Comment (benjefferies) [reply app/templates/summary.html]: I think you're right, @hamishtaplin also mentioned the same thing. I'm currently trying to figure out how 'answers' and 'items' fit in as 'SummaryItems'. Any suggestions are welcomed!

Comment (hamishtaplin) [reply app/templates/summary.html]: @benjefferies @iwootten Totally agree. Also, this logic isn't right as both of these things would need to be displayed anyway.

Comment (benjefferies) [reply app/templates/summary.html]: It's worth taking another look at the changes. I have minimised the logic in the template now although there is still a conditional check due to the dual nature of the date range question/answer

File: app/templates/summary.html [hunk 2]
```diff
@@ -30,51 +30,60 @@ <h2 class="header__subtitle mars">Please check carefully before submission.</h2>
 
         <dt class="summary__question summary__question--{{item.type}}" id="{{id}}">{{item.question | safe}}</dt>
         <dd class="summary__answer summary__answer--{{item.type}}">
+          {% for sub_item in item.sub_items %}
 
-          <div class="summary__answer-text" id="{{id}}-answer" data-qa="answer-{{outer_loop_index}}-{{inner_loop_index}} answer-type-{{item.type}}">
+            {% set item_id = id ~ "-" ~ loop.index0 %}
+            {% if item.sub_items|length > 1 %}
+              <dt class="summary__question summary__answer__title--{{sub_item.type}}" id="{{item_id}}">{{sub_item.answer_title | safe}}</dt>
+              <dd class="summary__answer summary__answer--{{item.type}}">
```
Comment (hamishtaplin) [inline app/templates/summary.html]: @benjefferies: Just a heads up: this isn't valid html as the `dt/dd` pair needs to be directly nested inside a `dl`. Also the `dd` tag isn't closed anywhere.

This bit of template no longer makes sense semantically anyway so I'm going to have to refactor the whole thing.

File: app/data/1_0112.json [hunk 3]
```diff
@@ -357,44 +320,33 @@
...
-                                            "validation": {
-                                                "messages": {
-                                                    "INTEGER_TOO_LARGE": "The maximum value allowed is 9999999999. Please correct your answer.",
-                                                    "NEGATIVE_INTEGER": "The value cannot be negative. Please correct your answer.",
-                                                    "NOT_INTEGER": "Please only enter whole numbers into the field.",
-                                                    "MANDATORY": "Please provide a value, even if your value is 0."
-                                                }
-                                            }
-                                        }
+                                              "display": {},
```
Comment (ajmaddaford) [inline app/data/1_0112.json]: The alignment of the JSON is a bit out (extra spaces)

File: tests/app/templating/summary/formatters/test_summary_item.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+from mock import Mock
+
+from app.templating.summary.summary_item import SummaryItem
+
+
+class TestSummaryItem(TestCase):
+
+    def test_create_summary_item(self):
+        # Given
+        schema = mock.MagicMock()
```
Comment (ajmaddaford) [inline tests/app/templating/summary/formatters/test_summary_item.py]: I think just calling this `schema` is confusing, `test_question_formatters.py` uses `question_schema`

File: tests/functional/spec/multiple-answers.spec.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    personalDetailsPage.setFirstName('Han')
+      .setSurname('Solo')
+      .submit()
+
+    // Then
+    expect(multipleAnswerSummaryPage.getFirstName()).to.equal('Han')
+    expect(multipleAnswerSummaryPage.getSurname()).to.equal('Solo')
+  })
+
+  it('Given I complete a survey has multiple answers for a question when I edit an answer then I appear on the page to edit my answer', function() {
```
Comment (ajmaddaford) [inline tests/functional/spec/multiple-answers.spec.js]: This doesn't read very well. I think it should either be 'Given I complete a survey **that** has...` or just drop the first bit and have 'Given a survey has multiple answers...'

Comment (benjefferies) [reply tests/functional/spec/multiple-answers.spec.js]: Good spot, fixed

File: tests/app/templating/summary/formatters/test_summary_item.py (hunk 4 above)
Comment (benjefferies) [reply tests/app/templating/summary/formatters/test_summary_item.py]: Good point, fixed.

File: app/data/1_0112.json (hunk 3 above)
Comment (benjefferies) [reply app/data/1_0112.json]: Good spot, fixed.

**Review by ajmaddaford — APPROVED** [review]
LGTM 👍

**Pull request merged**